gradle-modules/src/main/kotlin/org.sonarsource.cloud-native.rule-api.gradle.kts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SonarSource Cloud Native Gradle Modules
+ * Copyright (C) 2024-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+import org.sonarsource.cloudnative.gradle.RuleApiExtension
+import org.sonarsource.cloudnative.gradle.registerRuleApiGenerateTask
+import org.sonarsource.cloudnative.gradle.registerRuleApiUpdateTask
+import org.sonarsource.cloudnative.gradle.repox
+
+val ruleApi: Configuration = configurations.create("ruleApi")
+val ruleApiExtension = extensions.create<RuleApiExtension>("ruleApi")
+
+repositories {
+    repox("sonarsource-private-releases", providers, ruleApiExtension.fileOperations)
+    mavenCentral()
+}
+
+dependencies {
+    ruleApi("com.sonarsource.rule-api:rule-api:2.9.0.4061")
+}
+
+project.afterEvaluate {
```

Comment:
Nitpick. I guess it is hard to avoid the `afterEvaluate` usage? It is also ok to leave it as it is

Reply:
I thought it was not possible, because `ruleApiUpdate` has to depend on all tasks, and they will be available only after `ruleApiExtension` is configured. However, `dependsOn` can accept a `Provider`, so it is possible to avoid `afterEvaluate` here.

Reply:
Actually, it doesn't work; because of the lazy configuration `dependsOn` will be evaluated only if `ruleApiUpdate` will be called directly